apollo-router/src/plugins/telemetry/tracing/apollo_telemetry.rs

```diff
@@ -1,311 +1,487 @@
+                            sent_time: Some(span.start_time.into()),
+                            received_time: Some(span.end_time.into()),
+                        }),
+                    )),
+                })]
+            }
+            "flatten" => {
+                vec![TreeData::QueryPlan(QueryPlanNode {
+                    node: Some(apollo_spaceport::trace::query_plan_node::Node::Flatten(
+                        Box::new(FlattenNode {
+                            response_path: span
+                                .attributes
+                                .get(&APOLLO_PRIVATE_PATH)
+                                .and_then(Self::extract_string)
+                                .map(|v| {
+                                    v.split('/').filter(|v|!v.is_empty() && *v != "@").map(|v| {
+                                        if let Ok(index) = v.parse::<u32>() {
+                                            ResponsePathElement { id: Some(apollo_spaceport::trace::query_plan_node::response_path_element::Id::Index(index))}
+                                        } else {
+                                            ResponsePathElement { id: Some(apollo_spaceport::trace::query_plan_node::response_path_element::Id::FieldName(v.to_string())) }
+                                        }
+                                    }).collect()
+                                }).unwrap_or_default(),
+                            node: child_nodes
+                                .into_iter()
+                                .filter_map(|child| match child {
+                                    TreeData::QueryPlan(node) => Some(Box::new(node)),
+                                    _ => None,
+                                })
+                                .next(),
+                        }),
+                    )),
+                })]
+            }
+            "subgraph" => {
+                vec![TreeData::Trace(self.find_ftv1_trace(span))]
+            }
+            "supergraph" => {
+                //Currently some data is in the supergraph span as we don't have the a request hook in plugin.
+                child_nodes.push(TreeData::Supergraph {
+                    http: self.extract_http_data(span),
+                    client_name: span
+                        .attributes
+                        .get(&CLIENT_NAME)
+                        .and_then(Self::extract_string),
+                    client_version: span
+                        .attributes
+                        .get(&CLIENT_VERSION)
+                        .and_then(Self::extract_string),
+                    operation_signature: span
+                        .attributes
+                        .get(&APOLLO_PRIVATE_OPERATION_SIGNATURE)
+                        .and_then(Self::extract_string)
+                        .unwrap_or_default(),
+                });
+                child_nodes
+            }
+            "request" => {
```

Comment:
use the constant already created

Reply:
Fixed in [9c07453](https://github.com/apollographql/router/pull/1514/commits/9c074531a4a2cdf8b6a3c85683b175720f7933dd)